src/overlays/actors/ovl_En_Egol/z_en_egol.c

```diff
@@ -119,99 +316,1319 @@ const ActorInit En_Egol_InitVars = {
+            blockTo1.y = pos1.y - egBlock->world.pos.y;
+            blockTo1.z = pos1.z - egBlock->world.pos.z;
+            blockTo2.x = pos2.x - egBlock->world.pos.x;
+            blockTo2.y = pos2.y - egBlock->world.pos.y;
+            blockTo2.z = pos2.z - egBlock->world.pos.z;
+
+            dist1 = sqrtf(SQXZ(blockTo1));
+            dist2 = sqrtf(SQXZ(blockTo2));
+
+            if ((dist1 < 40.0f) || (dist2 < 40.0f)) {
+                dist1 = sqrtf(SQ(blockTo1.y));
+                dist2 = sqrtf(SQ(blockTo2.y));
+                if ((dist1 < 40.0f) || (dist2 < 40.0f)) {
+                    egBlock->colChkInfo.health = 0;
+                    Actor_PlaySfxAtPos(&this->actor, NA_SE_EV_WALL_BROKEN);
+                    break;
+                }
+            }
+        }
+        egBlock = egBlock->next;
+    }
+}
+
+void EnEgol_GetWaypoint(EnEgol* this) {
+    if ((this->pathIndex != -1) && (this->path != NULL) &&
+        !SubS_CopyPointFromPath(this->path, this->waypoint, &this->waypointPos)) {
+        Actor_MarkForDeath(&this->actor);
+    }
+}
+
+#define EYEGORE_SET_DIM(element, centerx, centery, centerz, radiusv, scalev) \
+    element.dim.modelSphere.radius = radiusv;                                \
+    element.dim.scale = scalev;                                              \
+    element.dim.modelSphere.center.x = centerx;                              \
+    element.dim.modelSphere.center.y = centery;                              \
+    element.dim.modelSphere.center.z = centerz
+
+void EnEgol_Init(Actor* thisx, PlayState* play2) {
+    PlayState* play = play2;
+    EnEgol* this = THIS;
+
+    this->actor.gravity = -2.0f;
+    Actor_SetScale(&this->actor, 0.015f);
+    this->actor.colChkInfo.damageTable = &sDamageTable;
+    ActorShape_Init(&this->actor.shape, 0.0f, ActorShadow_DrawCircle, 60.0f);
+    this->actor.colChkInfo.mass = MASS_IMMOVABLE;
+    this->actor.hintId = 0x4A;
+    this->actor.colChkInfo.health = 8;
+    SkelAnime_InitFlex(play, &this->skelAnime, &gEyegoreSkel, &gEyegoreStandAnim, this->jointTable, this->morphTable,
+                       EYEGORE_LIMB_MAX);
+    Collider_InitAndSetJntSph(play, &this->bodyCollider, &this->actor, &sJntSphInit1, this->bodyElements);
+    Collider_InitAndSetJntSph(play, &this->eyeCollider, &this->actor, &sJntSphInit2, this->eyeElements);
+    Collider_InitAndSetQuad(play, &this->laserCollider, &this->actor, &sQuatInit);
+    this->pathIndex = GET_EYEGORE_PATH(&this->actor);
+    if (this->pathIndex == 0x3F) {
+        this->pathIndex = -1;
+        Actor_MarkForDeath(&this->actor);
+        return;
+    }
+    this->wakeupRange = this->actor.world.rot.x * 20.0f;
+    this->actor.world.rot.x = 0;
+    if (this->wakeupRange < 0.0f) {
+        this->wakeupRange = 200.0f;
+    }
+    this->minLaserRange = this->actor.world.rot.z * 20.0f;
+    this->actor.world.rot.z = 0;
+    if (this->minLaserRange < 0.0f) {
+        this->minLaserRange = 200.0f;
+    }
```

Comment:
Been a while since I've dealt with an actor like this, but I believe we want getter macros for when an actor uses it rot values like params?

Reply:
I think yes, although these are continuous so not quite the same as discrete cases. I'd also add a blank line between the different types of checks.